src/sage/graphs/generators/random.py

```diff
@@ -1567,13 +1565,16 @@ def RandomPartialKTree(n, k, x, seed=None):
     g = RandomKTree(n,k,seed)
 
     # Check that x doesn't delete too many edges
-    if x > g.size():
-        raise ValueError("x must be less than the number of edges in the K-Tree with `n` nodes")
-    
-    for i in range(x):
-        randomEdge = g.random_edge()
-        g.delete_edge(randomEdge)
+    # This formula calculates how many edges are in `k`-tree with `n` nodes
+    if x > ((n*(n - 1)) / 2) + (n - k - 1) * k:
```

Comment:
The initial clique has k+1 nodes, not n.

Reply:
My mistake, fixed